src/main/java/com/example/gui/ChatboxViewBuilder.java

```diff
@@ -0,0 +1,245 @@
+import javafx.util.Builder;
+
+import java.io.File;
+import java.util.function.Consumer;
+
+public class ChatboxViewBuilder implements Builder<Region> {
+
+    private static final int PANEL_WIDTH = 640;
+
+    private final ChatboxModel model;
+    private final BooleanProperty inferenceRunning = new SimpleBooleanProperty(false);
+    private final Consumer<Runnable> actionHandler;
+
+    public ChatboxViewBuilder(ChatboxModel model, Consumer<Runnable> actionHandler) {
+        this.model = model;
+        this.actionHandler = actionHandler;
+    }
+
+    @Override
+    public Region build() {
+        VBox results = new VBox();
+
+        HBox panels = new HBox(createLeftPanel(), createRightPanel());
+        VBox.setVgrow(panels, Priority.ALWAYS);
+        results.getChildren().add(panels);
+
+        return results;
+    }
+
+    private Node createLeftPanel() {
+        VBox panel = new VBox(12);
+        panel.setPrefWidth(PANEL_WIDTH);
+        panel.setPadding(new Insets(24, 12, 24, 24));
+        HBox.setHgrow(panel, Priority.ALWAYS);
+        panel.getChildren().addAll(
+                createHeaderLabel("TornadoVM Chat"),
+                createEngineBox(),
+                createLlama3PathBox(),
+                createModelSelectBox(),
+                createLabel("Prompt:"),
+                createPromptBox(),
+                createRunButton(),
+                createLabel("Output:"),
+                createOutputArea()
+        );
+        return panel;
+    }
+
+    private Node createEngineBox() {
+        ComboBox<ChatboxModel.Engine> engineDropdown = new ComboBox<>();
+        engineDropdown.valueProperty().bindBidirectional(model.selectedEngineProperty());
+        engineDropdown.getItems().addAll(ChatboxModel.Engine.values());
+        engineDropdown.setMaxWidth(Double.MAX_VALUE);
+        HBox box = new HBox(8, createLabel("Engine:"), engineDropdown);
+        box.setAlignment(Pos.CENTER_LEFT);
+        return box;
+    }
+
+    private Node createLlama3PathBox() {
+        Button browseButton = new Button("Browse");
+        browseButton.getStyleClass().add(Styles.ACCENT);
+        browseButton.setMinWidth(80);
+        browseButton.disableProperty().bind(inferenceRunning);
+        browseButton.setOnAction(e -> {
+            DirectoryChooser dirChooser = new DirectoryChooser();
+            dirChooser.setTitle("Select GPULlama3.java Directory");
+            File selectedDir = dirChooser.showDialog(browseButton.getScene().getWindow());
+            if (selectedDir != null) {
+                model.setLlama3Path(selectedDir.getAbsolutePath());
+            }
+        });
+
+        TextField pathField = boundTextField(model.llama3PathProperty());
+        HBox box = new HBox(8, createLabel("Llama3 Path:"), pathField, browseButton);
+        box.setAlignment(Pos.CENTER_LEFT);
+        HBox.setHgrow(pathField, Priority.ALWAYS);
+        pathField.setMaxWidth(Double.MAX_VALUE);
+
+        return box;
+    }
```

Comment:
This box I think is not required as the GUI is laucnhed from within the application and it is running the ROOT directory of the project. So, no need to specify that.